clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -1215,6 +1241,69 @@ class ArchiveFileHandler final : public FileHandler {
   Error WriteBundle(raw_fd_ostream &OS, MemoryBuffer &Input) override {
     llvm_unreachable("unsupported for the ArchiveFileHandler");
   }
+
+private:
+  // NOTE: mostly a copy-paste of ReadHeader method.
+  Expected<std::vector<std::string>>
+  ReadTargetsFromChild(const Archive::Child &C) {
+    auto BinOrErr = C.getAsBinary();
+    if (!BinOrErr)
+      return BinOrErr.takeError();
+
+    auto &Bin = BinOrErr.get();
+    auto Obj = std::unique_ptr<ObjectFile>(cast<ObjectFile>(Bin.release()));
+    auto Buf = MemoryBuffer::getMemBuffer(Obj->getMemoryBufferRef(), false);
+    ObjectFileHandler OFH(std::move(Obj), BundlerConfig);
+    if (Error Err = OFH.ReadHeader(*Buf))
+      return Err;
+    Expected<std::optional<StringRef>> NameOrErr = OFH.ReadBundleStart(*Buf);
+    if (!NameOrErr)
+      return NameOrErr.takeError();
+
+    std::vector<std::string> Targets;
+    while (*NameOrErr) {
+      if (*NameOrErr)
+        Targets.emplace_back((**NameOrErr).str());
+      NameOrErr = OFH.ReadBundleStart(*Buf);
+      if (!NameOrErr)
+        return NameOrErr.takeError();
+    }
+
+    return Targets;
+  }
+
+  bool CheckTargetIsExcluded(StringRef Triple) {
+    // NOTE: "-sycldevice" Triple component has been deprecated.
+    // However, it still can be met in libraries that have been compiled before
+    // deprecation. For example, here Triple might be the following:
+    //  sycl-fpga_aoco-intel-unknown-sycldevice
+    //
+    // The workaround is to strip this Triple component if it is present.
+    Triple.consume_back("-sycldevice");
+    const auto &ExcludedTargetNames = BundlerConfig.ExcludedTargetNames;
+    auto it = std::find(ExcludedTargetNames.begin(), ExcludedTargetNames.end(),
+                        Triple);
+    return it != ExcludedTargetNames.end();
+  }
+
+  // Function reads targets from Child and checks whether one of Targets
+  // is in Excluded list.
+  Expected<bool>
+  CheckObjectFileContainsExcludedTargets(const Archive::Child &C) {
+    if (BundlerConfig.ExcludedTargetNames.empty())
+      return false;
+
+    auto TargetNamesOrErr = ReadTargetsFromChild(C);
+    if (!TargetNamesOrErr)
+      return TargetNamesOrErr.takeError();
+
+    auto TargetNames = TargetNamesOrErr.get();
+    for (const auto &TargetName : TargetNames)
+      if (CheckTargetIsExcluded(TargetName))
+        return true;
+
+    return false;
```

Comment:
Return `std::count_if() > 0` here as well? [cppreference](https://en.cppreference.com/w/cpp/algorithm/count)

Reply:
`CheckTargetIsExcluded` is a method. Thus, it can't be passed as UnaryPredicate. Also it can't be made static because it uses BundlerConfig. Trying to make it static and pass BundlerConfig through would make the code look ugly.